pallets/phala/src/pawnshop.rs

```diff
@@ -0,0 +1,352 @@
+			let active_stakes = Self::get_net_value(user.clone())?;
+			ensure!(
+				amount <= active_stakes,
+				Error::<T>::RedeemAmountExceedsTotalStake,
+			);
+			<T as mining::Config>::Currency::transfer(
+				&T::PawnShopAccountId::get(),
+				&user,
+				amount,
+				KeepAlive,
+			)?;
+			Self::burn_from(T::PPhaAssetId::get(), &user, amount)?;
+
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		#[frame_support::transactional]
+		pub fn vote(
+			origin: OriginFor<T>,
+			aye_amount: BalanceOf<T>,
+			nay_amount: BalanceOf<T>,
+			vote_id: ReferendumIndex,
+		) -> DispatchResult {
+			let user = ensure_signed(origin.clone())?;
+			if !Self::is_ongoing(vote_id) {
+				return Err(Error::<T>::ReferendumInvalid.into());
+			}
+
+			let active_stakes = Self::get_net_value(user.clone())?;
+			ensure!(
+				active_stakes >= aye_amount + nay_amount,
+				Error::<T>::VoteAmountLargerThanTotalStakes,
+			);
+			VoteAccountMap::<T>::insert(vote_id, &user, (aye_amount, nay_amount));
+			AccountVoteMap::<T>::insert(&user, vote_id, ());
+			let account_vote = Self::accumulate_account_vote(vote_id);
+			pallet_democracy::Pallet::<T>::vote(origin, vote_id, account_vote)?;
+			Self::update_user_locked(user)?;
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		#[frame_support::transactional]
+		pub fn unlock(
+			origin: OriginFor<T>,
+			vote_id: ReferendumIndex,
+			max_iterations: u32,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+			if Self::is_ongoing(vote_id) {
+				return Err(Error::<T>::ReferendumOngoing.into());
+			}
+			let mut iter = VoteAccountMap::<T>::iter_prefix(vote_id).drain();
+			let mut i = 0;
+			loop {
+				if let Some((user, _)) = iter.next() {
+					AccountVoteMap::<T>::remove(user.clone(), vote_id);
+					Self::update_user_locked(user.clone()).expect("useraccount should exist: qed.");
+					if i >= max_iterations {
+						break;
+					}
+					i += 1;
+				} else {
+					break;
+				}
+			}
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T>
+	where
+		BalanceOf<T>: sp_runtime::traits::AtLeast32BitUnsigned + Copy + FixedPointConvert + Display,
+		T: pallet_uniques::Config<CollectionId = CollectionId, ItemId = NftId>,
+		T: pallet_assets::Config<AssetId = u32, Balance = BalanceOf<T>>,
+		T: Config + vault::Config,
+	{
+		pub fn get_asset_id() -> u32 {
+			T::PPhaAssetId::get()
+		}
+
+		pub fn remove_dust(who: &T::AccountId, dust: BalanceOf<T>) {
+			debug_assert!(dust != Zero::zero());
+			if dust != Zero::zero() {
+				let actual_removed =
+					pallet_assets::Pallet::<T>::slash(T::PPhaAssetId::get(), who, dust)
+						.expect("slash should success with correct amount: qed.");
+				let (imbalance, _remaining) = <T as mining::Config>::Currency::slash(
+					&<mining::pallet::Pallet<T>>::account_id(),
+					dust,
+				);
+				T::OnSlashed::on_unbalanced(imbalance);
+				Self::deposit_event(Event::<T>::DustRemoved {
+					user: who.clone(),
+					amount: actual_removed,
+				});
+			}
+		}
+
+		pub fn mint_into(
+			asset_id: u32,
```

Comment:
No need to pass an `asset_id` to this function. It's just `T::PPhaAssetId::get()`.

Reply:
done